src/frames.ts

```diff
@@ -350,55 +341,64 @@ export class Frame {
   }
 
   async goto(url: string, options: GotoOptions = {}): Promise<network.Response | null> {
-    const headers = (this._page._state.extraHTTPHeaders || {});
-    let referer = headers['referer'] || headers['Referer'];
-    if (options.referer !== undefined) {
-      if (referer !== undefined && referer !== options.referer)
-        throw new Error('"referer" is already specified as extra HTTP header');
-      referer = options.referer;
-    }
-    url = helper.completeUserURL(url);
-
-    const frameTask = new FrameTask(this, options, url);
-    const sameDocumentPromise = frameTask.waitForSameDocumentNavigation();
-    const navigateResult = await frameTask.raceAgainstFailures(this._page._delegate.navigateFrame(this, url, referer)).catch(e => {
-      // Do not leave sameDocumentPromise unhandled.
-      sameDocumentPromise.catch(e => {});
-      throw e;
-    });
-    if (navigateResult.newDocumentId) {
-      // Do not leave sameDocumentPromise unhandled.
-      sameDocumentPromise.catch(e => {});
-      await frameTask.waitForSpecificDocument(navigateResult.newDocumentId);
-    } else {
-      await sameDocumentPromise;
-    }
-    const request = (navigateResult && navigateResult.newDocumentId) ? frameTask.request(navigateResult.newDocumentId) : null;
-    await frameTask.waitForLifecycle(options.waitUntil === undefined ? 'load' : options.waitUntil);
-    frameTask.done();
-    return request ? request._finalRequest().response() : null;
+    return Progress.runCancelableTask(async progress => {
```

Comment:
`this._page._runCancelableTask(() => {}, options)`

Reply:
I will follow up.